client/src/cmdlfem4x50.c

```diff
@@ -527,225 +633,684 @@ int em4x50_read(em4x50_data_t *etd, em4x50_word_t *out, bool verbose) {
         return PM3_ETIMEOUT;
     }
 
-    bool isOK = (resp.status & STATUS_SUCCESS) >> 1;
-    if (isOK == false) {
-        if (verbose)
-            PrintAndLogEx(FAILED, "reading " _RED_("failed"));
-
+    if (resp.status != PM3_SUCCESS)
         return PM3_ESOFT;
-    }
-
-    if (edata.pwd_given) {
-        bool login = resp.status & STATUS_LOGIN;
-        if (login == false) {
-            PrintAndLogEx(FAILED, "login failed");
-            return PM3_ESOFT;
-        }
-        PrintAndLogEx(SUCCESS, "login with password " _YELLOW_("%s"), sprint_hex_inrow(etd->password, 4));
-    }
 
     uint8_t *data = resp.data.asBytes;
     em4x50_word_t words[EM4X50_NO_WORDS];
-    if (edata.addr_given) {
-        prepare_result(data, etd->address, etd->address, words);
-    } else {
-        int now = (resp.status & STATUS_NO_WORDS) >> 2;
-        prepare_result(data, 0, now - 1, words);
-    }
+    prepare_result(data, etd->addresses & 0xFF, (etd->addresses >> 8) & 0xFF, words);
 
-    if (out != NULL) {
+    if (out != NULL)
         memcpy(out, &words, sizeof(em4x50_word_t) * EM4X50_NO_WORDS);
-    }
 
-    print_result(words, etd->address, etd->address);
+    print_result(words, etd->addresses & 0xFF, (etd->addresses >> 8) & 0xFF);
+
     return PM3_SUCCESS;
 }
 
 int CmdEM4x50Read(const char *Cmd) {
 
+    int pwdLen = 0;
+    int addr = 0;
+    uint8_t pwd[4] = {0x0};
     em4x50_data_t etd;
-    memset(&etd, 0x00, sizeof(em4x50_data_t));
 
-    etd.pwd_given = false;
+    // init
+    memset(&etd, 0x00, sizeof(em4x50_data_t));
     etd.addr_given = false;
-    etd.newpwd_given = false;
+    etd.pwd_given = false;
 
-    bool errors = false;
-    uint8_t cmdp = 0;
-    while (param_getchar(Cmd, cmdp) != 0x00 && !errors) {
+    CLIParserContext *ctx;
+    CLIParserInit(&ctx, "lf em 4x50_read",
+                  "Reads single EM4x50 block/word.",
+                  "lf em 4x50_read -b 3 -> reads block 3\n"
+                  "lf em 4x50_read -b 32 -p 12345678 -> reads block 32 with password 0x12345678\n"
+                 );
+
+    void *argtable[] = {
+        arg_param_begin,
+        arg_int1("b", "block", "<address>", "block/word address, dec"),
+        arg_str0("p", "passsword", "<password>", "password, hex, 4 bytes, lsb"),
```

Comment:
<password>   ->  <hex>

Reply:
Maybe a hint what you mean?

Reply:
I think github filtered out what looked like HTML tags from his comment, somehow they came through in the Discord integration.  `<password> -> <hex>` is what it said

Reply:
Ok, thanks.